<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ShaderTestFrameworkClass</class>
 <widget class="QMainWindow" name="ShaderTestFrameworkClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1180</width>
    <height>612</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ShaderTestFramework</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="My_OpenGLWidget" name="graphicDisplay">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>751</width>
      <height>571</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="updateButton">
    <property name="geometry">
     <rect>
      <x>900</x>
      <y>540</y>
      <width>141</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Update Shaders</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>10</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Vertex Shader</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>930</x>
      <y>240</y>
      <width>151</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Fragment Shader</string>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="vertText">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>30</y>
      <width>391</width>
      <height>201</height>
     </rect>
    </property>
    <property name="lineWrapMode">
     <enum>QPlainTextEdit::NoWrap</enum>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="fragText">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>260</y>
      <width>391</width>
      <height>271</height>
     </rect>
    </property>
    <property name="lineWrapMode">
     <enum>QPlainTextEdit::NoWrap</enum>
    </property>
   </widget>
   <widget class="QPushButton" name="downButton">
    <property name="geometry">
     <rect>
      <x>1070</x>
      <y>540</y>
      <width>101</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Down</string>
    </property>
   </widget>
   <widget class="QPushButton" name="leftButton">
    <property name="geometry">
     <rect>
      <x>780</x>
      <y>540</y>
      <width>91</width>
      <height>41</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate Left</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1180</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionLoad_Shaders"/>
    <addaction name="actionSave_Shaders"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionLoad_Shaders">
   <property name="text">
    <string>Load Shaders...</string>
   </property>
  </action>
  <action name="actionSave_Shaders">
   <property name="text">
    <string>Save Shaders...</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>My_OpenGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>my_openglwidget.h</header>
   <slots>
    <slot>rotateLeft()</slot>
    <slot>rotateDown()</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="shadertestframework.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>updateButton</sender>
   <signal>clicked()</signal>
   <receiver>ShaderTestFrameworkClass</receiver>
   <slot>updateShaders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>976</x>
     <y>563</y>
    </hint>
    <hint type="destinationlabel">
     <x>843</x>
     <y>245</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>leftButton</sender>
   <signal>clicked()</signal>
   <receiver>graphicDisplay</receiver>
   <slot>rotateLeft()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>822</x>
     <y>562</y>
    </hint>
    <hint type="destinationlabel">
     <x>659</x>
     <y>556</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>downButton</sender>
   <signal>clicked()</signal>
   <receiver>graphicDisplay</receiver>
   <slot>rotateDown()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>1109</x>
     <y>559</y>
    </hint>
    <hint type="destinationlabel">
     <x>674</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>ShaderTestFrameworkClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionLoad_Shaders</sender>
   <signal>triggered()</signal>
   <receiver>ShaderTestFrameworkClass</receiver>
   <slot>loadShaders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionSave_Shaders</sender>
   <signal>triggered()</signal>
   <receiver>ShaderTestFrameworkClass</receiver>
   <slot>saveShaders()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>589</x>
     <y>305</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>updateShaders()</slot>
  <slot>loadShaders()</slot>
  <slot>saveShaders()</slot>
 </slots>
</ui>
